etl/steps/data/garden/wid/2024-05-24/world_inequality_database.py

```diff
@@ -0,0 +1,371 @@
+
+# Initialize logger.
+log = get_logger()
+
+# Define combinations of variables to calculate relative poverty
+EXTRAPOLATED_DICT = {"no": "", "yes": "_extrapolated"}
+WELFARE_VARS = ["pretax", "posttax_nat", "posttax_dis", "wealth"]
+
+# Set table format when printing
+TABLEFMT = "pretty"
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("world_inequality_database")
+
+    # Read tables from meadow dataset.
+    tb = ds_meadow["world_inequality_database"].reset_index()
+    tb_percentiles = ds_meadow["world_inequality_database_distribution"].reset_index()
+    tb_fiscal = ds_meadow["world_inequality_database_fiscal"].reset_index()
+
+    #
+    # Process data.
+    # Change units and drop unnecessary columns
+    tb = drop_columns_and_transform(tb)
+
+    # Sanity checks
+    sanity_checks(tb)
+
+    ########################################
+    # Percentile data
+    ########################################
+
+    # Process data.
+    # Multiple share and share_extrapolated columns by 100
+    tb_percentiles[["share", "share_extrapolated"]] *= 100
+
+    # Multiply columns containing share in tb_fiscal by 100
+    tb_fiscal[list(tb_fiscal.filter(like="share"))] *= 100
+
+    # Add relative poverty values
+    tb_relative_poverty = add_relative_poverty(tb, tb_percentiles, EXTRAPOLATED_DICT, WELFARE_VARS)
+
+    # Merge tables
+    tb = pr.merge(tb, tb_relative_poverty, on=["country", "year"], how="left")
+
+    # Add metadata by code (key indicators)
+    tb = add_metadata_vars(tb)
+
+    # Add metadata by code (distributions)
+    tb_percentiles = add_metadata_vars_distribution(tb_percentiles)
+
+    # Set index and sort
+    tb = tb.format()
+    tb_percentiles = tb_percentiles.format(["country", "year", "welfare", "p", "percentile"])
+    tb_fiscal = tb_fiscal.format()
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset and add the garden table.
+    ds_garden = create_dataset(
+        dest_dir,
+        tables=[tb, tb_percentiles, tb_fiscal],
+        check_variables_metadata=True,
+        default_metadata=ds_meadow.metadata,
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+# Data processing function (cleaning and small transformations)
+def drop_columns_and_transform(tb: Table) -> Table:
+    """
+    Drop columns and transform shares by multiplying by 100
+    """
+    # Multiply shares by 100
+    tb[list(tb.filter(like="share"))] *= 100
+
+    # Delete age and pop, two one-value variables
+    tb = tb.drop(columns=["age", "pop", "age_extrapolated", "pop_extrapolated"])
+
+    # Delete some share ratios we are not using, and also the p0p40 (share) variable only available for pretax
+    drop_list = ["s90_s10_ratio", "s90_s50_ratio", "p0p40"]
+
+    for var in drop_list:
+        tb = tb[tb.columns.drop(list(tb.filter(like=var)))]
```

Comment:
Alternatively, I believe you could drop the irrelevant columns as follows:

```python
columns_drop = tb.filter(regex=r"|".join(drop_list)).columns
tb = tb.drop(columns=columns_drop)
```

(unsure if it is better, I just thought you could get rid of the for-loop)